applications/staex-iod.md

```diff
@@ -0,0 +1,326 @@
+
+## Project Overview :page_facing_up:
+
+### Overview
+
+Staex Internet of Data.
+
+The goal of this project is to create a Web3 IoT data infrastructure with a stable economy. In simple words we want to simplify Web3 onboarding process and bring more IoT device owners to share their useful data with some profit and other people to find and use such data transparently, securely and easily for their life or research.
+
+We want to bring Web3 data access infrastructure to your ecosystem. As a part of Web3 Foundation ecosystem we want to use Kusama for our smart contracts.
+
+Our team is interested in creating this project because Staex are working on several solutions for IoT devices, so we are fully interesting to propose such topics to community and improve IoT field at all with new solutions.
+
+### Project Details
+
+#### Architecture
+
+![architecture](https://raw.githubusercontent.com/lavrd/cdn/main/image.png)
+
+#### IoT devices indexing
+
+To give users possibility to find particular type of data or particular IoT device we need to index them somehow. Indexing part we want to implement through DIDs and Subsquid project.
+
+So we will implement ink! smart contract which will be deployed on Kusama network. This smart contract will have several methods:
+
+- Create DID
+- Add attribute
+- Get attribute
+- Update attribute
+- Delete attribute
+
+All these methods will produce smart contract events.
+
+To index and retrieve IoT devices (DID) we want to use Subsquid project and its API as it index all transactions and events from blockchain network and we can query them using GraphQL API from our UI. User can use public indexer or start its own. Subsquid already has an integration with Kusama.
+
+To make process of setup or update DID for IoT device easier, we will implement such logic in our provisioner. So user will need just to set some fields in config file and provisioner will sync DID attributes with network through smart contracts calls.
+
+Attributes which we want to store in DIDs:
+
+- Type of data - type of data which IoT device produces, it can be temperature, humidity or something like CCTV camera output, etc.
+- Location - it can be coordinates or name of the place.
+- Public key - device public key. We will need it later to create shared secret or know exact files location in IPFS.
+- Price to access data per KB.
+- Price to pin data per hour.
+- Device can include some other attributes but it is optional.
+
+After that we will implement a UI where users can see devices and find necessary one through filters. UI will be quite simple: header/footer + a table with device list, each row will have an information from DID. Plus we need an input or select field for filters.
+
+#### ACL
+
+Access control layer is here to restrict data usage from users which didn't pay for a data.
+
+And here is a flow: 
+
+1. User puts some tokens to smart contract for exact owner and point which type of data it needs.
```

Comment:
How would the user send tokens to the smart contract? Would they need to use XCM first to teleport the tokens to the chain that is hosting said contracts?

Reply:
I think in next project phases we will implement this part like this:
- First time users will need to use UI to buy data access with particular tokens like SDN/SBY.
- In future for mass adoption we can implement a feature for UI to sell/buy data access using different tokens from different chains. For example IoT device owner wants to sell data for USDT, but a person wants to buy an access using DOT. Or SDN <-> KSM. So we can implement in UI auto teleport/exchange for user. As I understand it is possible through XCM and Polkadot.js.